From de15447057bf84535959cb8195b7bc5e2f4ed4bf Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Sun, 9 Jan 2011 23:42:24 +0800
Subject: [PATCH 09/10] usb: allow user specified USB endpoint size

Patch comes from backfire_10.03
  backfire/target/linux/generic-2.6/patches-2.6.32:
    801-usb_serial_endpoint_size.patch

Integrated-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/usb/serial/usb-serial.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/serial/usb-serial.c b/drivers/usb/serial/usb-serial.c
index 6954de5..6405a29 100644
--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -61,6 +61,7 @@ static struct usb_driver usb_serial_driver = {
    drivers depend on it.
 */
 
+static ushort maxSize;
 static int debug;
 /* initially all NULL */
 static struct usb_serial *serial_table[SERIAL_TTY_MINORS];
@@ -913,7 +914,7 @@ int usb_serial_probe(struct usb_interface *interface,
 		}
 		buffer_size = serial->type->bulk_in_size;
 		if (!buffer_size)
-			buffer_size = le16_to_cpu(endpoint->wMaxPacketSize);
+			buffer_size = (endpoint->wMaxPacketSize > maxSize) ? endpoint->wMaxPacketSize : maxSize;
 		port->bulk_in_size = buffer_size;
 		port->bulk_in_endpointAddress = endpoint->bEndpointAddress;
 		port->bulk_in_buffer = kmalloc(buffer_size, GFP_KERNEL);
@@ -1388,3 +1389,5 @@ MODULE_LICENSE("GPL");
 
 module_param(debug, bool, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(debug, "Debug enabled or not");
+module_param(maxSize, ushort, 0);
+MODULE_PARM_DESC(maxSize, "User specified USB endpoint size");
-- 
1.7.0.4

